protocol.md

```diff
@@ -317,8 +317,47 @@ The value of the  `Upload-Expires` header MUST be in
 
 ### Checksums
 
-This extension will define how to provide per file or per chunk checksums for
-uploaded files.
+Clients and servers MAY implement and use this extension to verify data
+integrity per chunk. In this case the server MUST add the `checksum` element to
+the `TUS-Extension` header.
+
+A client MAY include the `Content-MD5` header and its appropriate value in a
```

Comment:
Refer to RFC for what is meant by "proper value"? It's actually quite tricky, IMHO, and may result in incompatible implementations. People may skip the example and assume it should be the hex-string with the checksum. Others may use base64 of that. Content-Transfer-Encoding needs to be taken into consideration, etc.

The rest looks great.